py/sentry_streams/flink/flink_adapter.py

```diff
@@ -53,3 +57,22 @@ def sink(self, step: Step, stream: Any) -> Any:
         )
 
         return stream.sink_to(sink)
+
+    def map(self, step: Step, stream: Any) -> Any:
+
+        assert hasattr(step, "function")
+        fn_path = step.function
+        mod, cls, fn = fn_path.rsplit(".", 2)
+
+        try:
+            module = get_module(mod)
+
+        except ImportError as e:
+            raise e
+
+        imported_cls = getattr(module, cls)
+        imported_fn = getattr(imported_cls, fn)
+
+        # The output type must be specified
+        # TODO: Remove hardcoded output type
```

Comment:
Are you sure it has to be specified ? 
I created a streaming app that does not specify that 

https://github.com/getsentry/streaming-tests/blob/main/pyflink_perf/flink_perf.py#L92-L103

Reply:
So the default return type of Map is pickled byte array https://nightlies.apache.org/flink/flink-docs-release-1.18/api/python/reference/pyflink.datastream/api/pyflink.datastream.data_stream.DataStream.map.html?highlight=map#pyflink.datastream.data_stream.DataStream.map 

The output type for Map must be specified in cases where there is a next step after which does some sort of serialization. In this case, the next step is a Sink which uses the `SimpleStringSchema` to serialize strings into bytes objects for Kafka. 

Without the output type here, the job execution fails because Flink is not able to recognize or convert the output of Map as a Java String (got a bunch of errors about failing to cast to String). 

Anyways, I'll add this as a comment elsewhere